de.metas.business/src/main/java/de/metas/pricing/service/impl/PricingBL.java

```diff
@@ -113,18 +113,22 @@ public IEditablePricingContext createInitialContext(
 		pricingCtx.setBPartnerId(bPartnerId);
 		pricingCtx.setConvertPriceToContextUOM(true); // backward compatibility
 
+
+		if(quantity == null || quantity.signum() == 0)
+		{
+			pricingCtx.setQty(BigDecimal.ONE);
+		}
+
 		if (quantity != null)
 		{
 			if (quantity.signum() != 0)
 			{
 				pricingCtx.setQty(quantity.toBigDecimal());
 			}
+
 			pricingCtx.setUomId(quantity.getUomId());
 		}
```

Comment:
```suggestion
		if (quantity == null)
		{			
			pricingCtx.setQty(ONE);
		} 
		else 
		{
			pricingCtx.setQty(quantity.isZero() ? ONE: quantity.toBigDecimal());
			pricingCtx.setUomId(quantity.getUomId());
		}
```

if think this should be equivalent, pls check. at any rate, i think it's clearer